codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/ApplicationTransport.java

```diff
@@ -98,23 +103,26 @@ public static ApplicationTransport createDefaultHttpApplicationTransport() {
 
         ClientConfig wireTrace = (new ClientConfig.Builder())
                 .name("http_wire_trace")
-                .type("bool")
+                .type("Boolean")
                 .defaultValue("false")
                 .documentation("Enable debug wire trace on http requests.")
+                .allowOperationOverride()
                 .build();
 
         ClientConfig logger = (new ClientConfig.Builder())
                 .name("logger")
                 .type("Logger")
-                .defaultValue("stdout")
-                .initializationCustomization(
-                        "@logger = options.fetch(:logger, Logger.new($stdout, level: @log_level))")
+                .documentationDefaultValue("$stdout")
+                .defaults(new ConfigProviderChain.Builder()
+                        .dynamicProvider("proc { |cfg| Logger.new($stdout, level: cfg[:log_level]) } ")
+                        .build()
+                )
                 .documentation("Logger to use for output")
                 .build();
 
         ClientConfig logLevel = (new ClientConfig.Builder())
                 .name("log_level")
-                .type("symbol")
+                .type("Symbol")
```

Comment:
I worry a little bit about the String usage here still.  Since we have code that requires exact values (and missing an uppercase woudl be easy).  Lets at least define constants on the Java side?  (or we could change the builder to have methods like `.SymbolType()`.  What do you think?